graphql/appointment/mutations.ts

```diff
@@ -0,0 +1,227 @@
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
+import { Lecture as Appointment } from '../generated/models';
+import { Role } from '../../common/user/roles';
+import {
+    AppointmentCreateGroupInput,
+    AppointmentCreateInputFull,
+    AppointmentCreateMatchInput,
+    createAppointments,
+    createGroupAppointment,
+    createGroupAppointments,
+    createMatchAppointment,
+    createMatchAppointments,
+} from '../../common/appointment/create';
+import { getSessionUser } from '../authentication';
+import { GraphQLContext } from '../context';
+import { AuthorizedDeferred, hasAccess } from '../authorizations';
+import { prisma } from '../../common/prisma';
+import { getLecture } from '../../graphql/util';
+import { getLogger } from 'log4js';
+import { Field, InputType, Int } from 'type-graphql';
+import { lecture_appointmenttype_enum } from '@prisma/client';
+import { getUserType } from '../../common/user';
+
+const logger = getLogger('MutateAppointmentsResolver');
+
+const getOrganizersStudentId = (context: GraphQLContext) => {
+    const { studentId } = getSessionUser(context);
+    if (!studentId) {
+        throw new Error('Only students can be organizers');
+    }
+    return studentId;
+};
+
+const mergeOrganizersWithSessionUserId = (organizers: number[] = [], context: GraphQLContext) => {
+    const studentId = getOrganizersStudentId(context);
+    if (!studentId) {
+        return organizers; // only Students can be organizers
+    }
+    if (organizers.includes(studentId)) {
+        return organizers; // already in organizers
+    }
+    return [...organizers, studentId];
+};
+
+@InputType()
+class AppointmentUpdateInput {
+    @Field(() => Int)
+    id: number;
+    @Field(() => String, { nullable: true })
+    title?: string;
+    @Field(() => String, { nullable: true })
+    description?: string;
+    @Field(() => Date)
+    start: Date;
+    @Field(() => Int)
+    duration: number;
+}
+
+@Resolver(() => Appointment)
+export class MutateAppointmentResolver {
+    @Mutation(() => Boolean)
+    @Authorized(Role.ADMIN, Role.STUDENT)
+    async appointmentCreate(@Ctx() context: GraphQLContext, @Arg('appointment') appointment: AppointmentCreateInputFull) {
```

Comment:
So I can just create an appointment with a subcourseId or matchId even though I'm neither a subcourse instructor nor on the Match, also passing in any participant ids without validation? What's the usecase?

Reply:
This is meant as a workaround vor admins to create appointments via the api for pupils, students or even screeners. 
This will not be the api used in the user app.